src/bootstrap/llvm.rs

```diff
@@ -215,6 +215,23 @@ pub(crate) fn is_ci_llvm_available(config: &Config, asserts: bool) -> bool {
     true
 }
 
+pub(crate) fn get_llvm_opts_from_ci(config: &Config) -> Option<LLVMConfig> {
+    if config.dry_run() || is_ci_llvm_modified(config) {
+        return None;
+    }
+
+    let llvm_sha = detect_llvm_sha(config, config.rust_info.is_managed_git_subrepository());
+    config.download_ci_llvm_opts(&llvm_sha);
+
+    let config_path = config.ci_llvm_root_opts().join("llvm-opts.json");
+    let Ok(config) = serde_json::from_slice::<LLVMConfig>(&t!(std::fs::read(config_path))) else {
+        // The LLVM config has changed its format or is corrupted in some way
+        eprintln!("Cannot deserialize LLVM config from llvm-opts.json");
+        return None;
```

Comment:
hmm, when would this happen? continuing the build when we can't parse the format seems unfortunate, i worry it'll silence a real bug.

Reply:
For example, if the struct is modified in a backwards incompatible way (e.g. new option added/removed), then it may fail to parse. This situation necessarily means that the config has changed somehow, and therefore we shouldn't reuse the LLVM from CI.

In theory, we could lose all caching if there was some bug and it was never deserialized properly, but the alternative is basically failing the build if the format changes, which we probably shouldn't do?

Reply:
Or we could devise some versioning scheme for the config (Protobuf? :laughing: ), but that seems like overkill.